File: source/downloads/code/LeetCode/Python/Jump-Game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        #index_true: current reachable index.
        if not nums or len(nums) == 0:
            return False

        index_true = len(nums)-1
        for i in range(len(nums)-1,-1,-1):
            if nums[i]+i >= index_true:
                index_true = i
        if index_true == 0:
            return True
        else:
            return False
        

File: downloads/code/LeetCode/Java/Jump-Game.java
public class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        if(n==0) return false;
        int index_true = n-1;
        for(int i=n-2; i>=0; i--){
            if(i+nums[i] >= index_true) 
                index_true = i;
        
        }
        return index_true==0;
                
    }
}
